buildSrc/src/main/groovy/com/github/okbuilds/okbuck/generator/BuckFileGenerator.groovy

```diff
@@ -51,6 +53,34 @@ final class BuckFileGenerator {
                     break
             }
         }
+
+        switch (ProjectUtil.getType(project)) {
```

Comment:
just use the exisitng rules functions for each target and add a new rule in each. No need to switch case again